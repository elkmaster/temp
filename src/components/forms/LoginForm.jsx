import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
// TODO: validation
export const LoginForm = ({ handleSubmit }) => (
  <form onSubmit={handleSubmit}>
    <Field
      name="email"
      component="input"
      type="email"
      placeholder="email"
    />
    <Field
      name="password"
      component="input"
      type="text"
      placeholder="password"
    />
    <button type="submit">Login</button>
  </form>
);

LoginForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

export default reduxForm({
  form: 'loginForm', // a unique identifier for this form
  touchOnBlur: false,
})(LoginForm);
