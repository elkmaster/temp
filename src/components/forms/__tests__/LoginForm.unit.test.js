import React from 'react';
import { shallow,  } from 'enzyme';
import faker from 'faker';

import { LoginForm } from '../LoginForm';

const setup = (propsOverrides) => {
  const props = Object.assign({
      handleSubmit: jest.fn(),
    },
    propsOverrides);

  const wrapper = shallow(<LoginForm {...props} />);

  return {
    props,
    wrapper,
  };
};

describe('LoginForm', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  //TODO: check if there is more optimal way to query field types????
  it('should has correct input fields', () => {
    const { wrapper } = setup();
    expect(wrapper.findWhere(n => n.props().type == 'email')).toHaveLength(1);
    expect(wrapper.findWhere(n => n.props().type == 'password')).toHaveLength(1);
  });

  describe('form submit', () => {
    it('should call handleSubmit on submit', () => {
      const { wrapper, props } = setup();
      wrapper.simulate('submit');
      expect(props.handleSubmit).toBeCalled();
    });

    it('should call handleSubmit with correct params', () => {
      const { wrapper, props } = setup();
      const payload = {
        values: {
          email: faker.internet.email(),
          password: faker.internet.password(),
        },
      };
      wrapper.simulate('submit', payload);

      expect(props.handleSubmit).toBeCalledWith(payload);
    });
  });

});
