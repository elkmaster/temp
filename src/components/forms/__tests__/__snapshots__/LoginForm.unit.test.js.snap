// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component="input"
    name="email"
    placeholder="email"
    type="email"
  />
  <Field
    component="input"
    name="password"
    placeholder="password"
    type="password"
  />
  <button
    type="submit"
  >
    Login
  </button>
</form>
`;
